### ISSUE TITLE
Insulator Health Classification NN Training

### ISSUE THREAD

**Issue by stephenwang5** [kite-capstone/meishur-mission#2]
We selected YOLOv8 for detecting and classifying insulators because YOLO is a good architecture for localizing potential objects in an image, then use it to make more informed classification result, compared to classifying the whole image without segmentation or two separate NNs to do these two tasks. There's a [compelling graph from Ultralytics](https://raw.githubusercontent.com/ultralytics/assets/main/yolov8/yolo-comparison-plots.png) that says YOLOv8 is optimized for inference time and accuracy.

We have found a handful of datasets of insulators in healthy/broken conditions. I think it's worth trying to

1. Fine-tune the Roboflow model using our dataset(s)
2. Use transfer learning from a general-purposed, pretrained model to insulators

Then compare the effectiveness of the existing Roboflow model(s) and the one(s) we train.

This roughly requires the following tasks

- [ ] segment the test dataset into validation and test
- [ ] append a fully-connected layer, or otherwise modify an existing general pretrained model
- [ ] append a fully-connected layer, or otherwise modify an existing Roboflow model
- [ ] train and compare the test set performance after validation is good

**Comment by niobiumneo** [comment]
kinda done closing now
